### PR TITLE
[ALLUXIO-2515]Show inconsistent path list on WebUI 

### PR DESCRIPTION
**Pull request by Yiran-wu**
#Show inconsistent path list on WebUI
https://alluxio.atlassian.net/browse/ALLUXIO-2515
Normal
image

Error(inconsistent files)
image

### CODE REVIEW

File: core/server/src/main/webapp/general.jsp [hunk 1]
```diff
@@ -13,6 +13,7 @@
 <%@ page import="java.util.*" %>
 <%@ page import="alluxio.web.*" %>
 
+
```
Comment (maobaolong) [inline core/server/src/main/webapp/general.jsp]: can you delete this blank line.

File: core/server/src/main/webapp/general.jsp [hunk 2]
```diff
@@ -114,6 +115,40 @@
...
+        </div>
+        <div id="data4" class="accordion-body collapse in">
+          <div class="accordion-inner">
+            <table class="table table-hover table-condensed">
+              <thead>
+              <th>Path List </th>
+              </thead>
+              <tbody>
+              <%
+                java.util.List array = (java.util.List) request.getAttribute("inconsistentPathItems");
```
Comment (maobaolong) [inline core/server/src/main/webapp/general.jsp]: your jsp head has bean import List, so you can use it by class name only.

**Comment by maobaolong** [comment]
LGTM. thanks for the contribution. Looking forward to seeing you more in our community.

File: core/server/src/main/webapp/general.jsp [hunk 3]
```diff
@@ -114,6 +114,38 @@
...
     </div>
   </div>
+  <!--  show inconsistent path  -->
+  <% if ((Integer) request.getAttribute("inconsistentPaths") != 0) { %>
+  <div class="row-fluid">
+    <div class="accordion span14" id="accordion4">
+      <div class="accordion-group">
+        <div class="accordion-heading">
+          <a class="accordion-toggle" data-toggle="collapse" data-parent="#accordion4" href="#data4">
+            <h4>Inconsistent Files</h4>
```
Comment (calvinjia) [inline core/server/src/main/webapp/general.jsp]: Could you clarify this is only for the startup check and may not be up to date? Also that the user can get the updated status by restarting Alluxio?

Comment (Yiran-wu) [inline core/server/src/main/webapp/general.jsp]: need to restart the alluxio master server for the latest status.
We have two options:
1. We can add some information in the WebUI, tell the user "need to restart to get the latest information"
2.Refresh WebUI get lastest status.  Add a thread in the background real-time check, but this method requires frequent calls "FileSystemMaster.checkConsistency ()", the internal implementation of this function will affect the performance of alluxio. At the same time due to frequent internal lock INode, may lead to deadlock.

We can use the first way to prompt the user. Later to add more advanced refresh function.
what do you think?

Comment (calvinjia) [inline core/server/src/main/webapp/general.jsp]: I fully agree, with going for option 1 in this PR. For the more advanced function of refresh, this ticket is already open: https://alluxio.atlassian.net/browse/ALLUXIO-2411.

File: core/server/src/main/webapp/general.jsp [hunk 4]
```diff
@@ -114,6 +114,38 @@
...
+        <div class="accordion-heading">
+          <a class="accordion-toggle" data-toggle="collapse" data-parent="#accordion4" href="#data4">
+            <h4>Inconsistent Files Details</h4>
+          </a>
+        </div>
+        <div id="data4" class="accordion-body collapse in">
+          <div class="accordion-inner">
+            <table class="table table-hover table-condensed">
+              <thead>
+                <th> <font color="red">On Startup, <%= request.getAttribute("inconsistentPaths") %> inconsistent files were fount. The following files may be corrupted: </font></th>
```
Comment (calvinjia) [inline core/server/src/main/webapp/general.jsp]: `fount` -> `found`, could you also include some information about this check not updating until the system is restarted?

Comment (Yiran-wu) [inline core/server/src/main/webapp/general.jsp]: @calvinjia  ,  Hi,  I've fixed it.

image

**Comment by calvinjia** [comment]
LGTM

**Comment by calvinjia** [comment]
@Yiran-wu Thanks for the contribution!

**Pull request merged**